lib/Ebook.php

```diff
@@ -1651,6 +1726,9 @@ public function Save(): void{
 
 		$this->RemoveTocEntries();
 		$this->AddTocEntries();
+
+		$this->RemoveEbookPlaceholder();
```

Comment:
Let's rethink how we do this, because we're adding two DB calls for every call to `Ebook::Save()` even if no changes were made to the placeholder. Saving a full ebook is going to be much more common than saving a placeholder so this will be a big drag.

One approach could be to pass a parameter to `Ebook::Save()` indicating if we want to save the placeholder as well. Maybe `Ebook::Save(bool $updatePlaceholder = false)`. That way a regular update won't make these DB calls but we can specify in for example `PATCH /ebook-placeholders` that we actually want to update the placeholder.

Reply:
Good point that full ebooks will be much more common than saving a placeholder, but I'll push back that this will add two DB calls and be a big drag.

For full ebooks, there will be just one DB call:

```  
DELETE from EbookPlaceholders  
where EbookId = ?  
```  

This is similar to what we do for `TocEntries`. Only about 13% of ebooks have `TocEntries`, but every `Save()` calls:

```  
DELETE from TocEntries  
where EbookId = ?  
```  

On my underpowered machine, I can't tell the difference between calling `RemoveEbookPlaceholder()/AddEbookPlaceholder()` and not when calling `Save()` on 1000 full ebooks (via `deploy-ebook-to-www --no-build` etc.). It takes about 45 seconds either way.

I'm going to leave this as-is unless you feel strongly about it.